proto/ptera/v1/ptera.proto

```
@@ -12,6 +12,17 @@ service PteraService {
   rpc CompleteCard(CompleteCardRequest) returns (CompleteCardResponse);
 }
 
+service BattleService {
+  rpc StartBattle(StartBattleRequest) returns (StartBattleResponse);
+  rpc Attack(AttackRequest) returns (AttackResponse);
+  rpc Retreat(RetreatRequest) returns (RetreatResponse);
+
+  // Battle Request (Matching) RPCs
+  rpc SendBattleRequest(SendBattleRequestRequest) returns (BattleRequest);
```

Buf (1.61.0)

"ptera.v1.BattleRequest" is used as the request or response type for multiple RPCs.

(RPC_REQUEST_RESPONSE_UNIQUE)